src/wallet/electrum.cpp

```diff
@@ -533,18 +534,29 @@ long_hash decode_mnemonic(const word_list& mnemonic,
         hmac_iterations);
 }
 
-bool validate_mnemonic(const word_list& mnemonic, const dictionary& lexicon,
-    seed prefix)
+bool validate_mnemonic(const word_list& mnemonic, const data_chunk& entropy,
+    const dictionary& lexicon, seed prefix)
 {
+    // To validate a specified wordlist given some starting entropy,
+    // we create a new one and ensure that the passed in mnemonic
+    // decodes to the same final entropy as the original mnenmonic.
+    //
+    // Creating a mnemonic is self validating, in that we guarantee
+    // that the entropy used to create it decodes back to that same
+    // entropy value.  Validation ensures that the passed in mnemonic
+    // and entropy is consistent by verifying a match on re-creation.
+    const auto created_mnemonic = create_mnemonic(entropy, lexicon, prefix);
+
     return is_new_seed(mnemonic, get_seed_prefix(prefix)) &&
-        (mnemonic_decode(mnemonic, lexicon) == 0);
+        (mnemonic_decode(mnemonic, lexicon) == mnemonic_decode(created_mnemonic, lexicon));
```

Comment:
Note that we can't just do this here:

```C++
 return is_new_seed(mnemonic, get_seed_prefix(prefix)) &&
        (mnemonic_decode(mnemonic, lexicon) == entropy);
```

The reason is because the starting entropy does not have to equal the final entropy.  The electrum algorithm advances it in order to create the mnemonic, so the best way to validate it seems to tie it to a newly generated one as this PR does.